core/src/main/java/tech/tablesaw/joining/DataFrameJoiner.java

```diff
@@ -683,10 +849,13 @@ private void crossProduct(
       Table table2,
       Selection table1Rows,
       Selection table2Rows,
-      Set<Integer> ignoreColumns) {
+      Set<Integer> ignoreColumns,
+      boolean keepTable2JoinKeyColumns) {
     for (int c = 0; c < table1.columnCount() + table2.columnCount(); c++) {
-      if (ignoreColumns.contains(c)) {
-        continue;
+      if (!keepTable2JoinKeyColumns) {
```

Comment:
you could combine these `if` checks into one (same below):

```
if (!keepTable2JoinKeyColumns && ignoreColumns.contains(c)) {
  continue;
}
```

Reply:
OK, I have done it.:)